### DISCUSSION TITLE
Welcome to cosi-data-challenge-2 Discussions!

### DISCUSSION THREAD

**Discussion by zoglauer** [cositools/cosi-data-challenge-2#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ags-thomas** [comment]
Hello my name is Gus Thomas, I am a CS and Physics undergrad at Wooster! I am working at WashU this summer, and I'm interested in proposing help for this project as a senior independent study :))

I have experience with Python, and some experience with data science. I'm helping out with GRB localization on the ADAPT mission right now. I'm focused on using machine learning methods for deta estimation and background emission classification. I don't know all of the details yet though, I'm only an undergrad!

I have a few questions:
1. Does this project have any long-term goals of moving the data processing pipeline to chip? At ADAPT we are limited by 5 s latency to ground, so a lot of our work is focused on doing reconstruction on chip to capture faster events. It's my understanding that conventional satellites send raw observations to ground for processing though.
2. Do you have a need for data processing acceleration? It seems like many of the questions here focus on accuracy.
3. Is the intention of the October meeting to share results for this year's challenge, or more so to get accustomed to the collaborations' needs?

**Reply by israelmcmc** [reply]
Hi @augustus-thomas! Thanks for reaching out. Here's my take on your questions, others can chime in:

1. Probably not. Most of the reconstruction and processing will be performed on the ground. This is because we don't have requirements that call for low on-board latency --such as slewing-- and the transient triggering will be based on overall rates --either from the veto shields or the germanium detector. There have been ideas floating around to have a simple on-board algorithm running on a chip to pre-select the event downlinked to the ground for reconstructions, but currently, they are very simple, like selecting only 2-hits events.
2. We're very much in need of data processing acceleration! Our current understanding is that the bulk of our current discrepancies come from having a detector response that is too coarse. If we can improve the handling of the response, we can get more accurate for sure.
3. It's mostly going to be a hands-on workshop towards the next data challenge.

There are plenty of things to do if you want to get involved, cosipy is still at an early stage. In particular, we haven't profiled most of our code, so it's likely there are a bunch of low-hanging fruits. Identifying the bottleneck systematically would be the place to start. If you are still interested, we can keep discussing in the corresponding issue: https://github.com/cositools/cosipy/issues/27
